python/lib/dependabot/python/update_checker.rb

```diff
@@ -252,6 +252,8 @@ def updated_version_req_lower_bound
       end
 
       def fetch_latest_version
+        return nil if PoetryVersionResolver.path_dependency?(pyproject.content, dependency)
```

Comment:
Should this also check if the `resolver_type` is poetry and/or `pyproject` exists? Looks like this will be called for all python ecosystems where this file might not exist so would call `content` on nil.

Or what happens if we don't return early and rely on the resolver returning nil?

Reply:
> What happens if we don't return early and rely on the resolver returning nil?

Ah just spotted your comment about this.